Source/WebCore/loader/MixedContentChecker.cpp

```diff
@@ -68,6 +70,11 @@ bool MixedContentChecker::canDisplayInsecureContent(Frame& frame, SecurityOrigin
     if (!frame.document()->contentSecurityPolicy()->allowRunningOrDisplayingInsecureContent(url))
         return false;
 
+    if (isWhitelisted(securityOrigin.toString(), url.protocolHostAndPort())) {
+        logWarning(frame, true, "display"_s, url);
+        return true;
+    }
+
```

Comment:
I'm curious about why you added this check here, cause it means that the whitelisting feature won't work if allowDisplayingInsecureContent of allowRunningInsecureContent are false. In my mind the whitelisting feature should work independently of those settings. Even more, the whitelist is needed to allow displaying and running insecure content when settings are off, otherwise the access will be denied.

Reply:
Yes, you're right, this condition should be moved above the allowRunningOrDisplayingInsecureContent condition